src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -299,6 +301,7 @@ internal interface ISyntaxFactsService : ILanguageService
         bool IsElementAccessExpression(SyntaxNode node);
         bool IsIndexerMemberCRef(SyntaxNode node);
         SyntaxNode GetContainingPropertyDeclaration(SyntaxNode node);
+        SyntaxNode GetContainingParameter(SyntaxNode node);
```

Comment:
are these two helpers just calls to `GetAncestorOrThis<TWhatever>()`?    If so, i would remove.  I don't think they have value.

Reply:
I can lift the code to the RefactoringHelpers service (respertively their lang. specific implementations) but I need it to be able to have the extractor lang. agnostic.

Reply:
isn't refactoringhelperservice abstract?  you can templatize it with the node types you care about (like you've done with specific features).  The C# and VB subclasses then just derive from the right instantiation of that type.  it's nifty :)

Reply:
I've tried to remain as generic as possible but you're right, it's probably better to go that route. Will investigate and see.